### ISSUE TITLE
For loop should dereference byref<'a> to 'a, just as Span's indexer does

### ISSUE THREAD

**Issue by vasily-kirichenko** [Microsoft/visualfsharp#5269]
![image](https://user-images.githubusercontent.com/873919/42125686-19ad5670-7c84-11e8-8513-d26492098355.png)

but

![image](https://user-images.githubusercontent.com/873919/42125693-3048d418-7c84-11e8-8273-0392eb6ccd10.png)

**Comment by majocha** [comment]
On the other hand it enables in-place mutability.

**Comment by vasily-kirichenko** [comment]
Yeah. Interesting, but inconsistent.

**Comment by TIHan** [comment]
Yes, this is a bug it looks like.

**Comment by TIHan** [comment]
Note, you can do something like this in C#:
```csharp
using System;
public class C {
    public void M() {
        var s = Span<int>.Empty;
        
        foreach(ref int x in s)
        {
            x = 5;
            Console.WriteLine(x);
        }
    }
}
```

In F#, should it look like this?
```fsharp
let spanSum (data: Span<byte>) =
    for &x in data do
        ()
```

**Comment by dsyme** [comment]
Great find, I didn't spot this `foreach` in the C# spec.  

I agree with the change in #5285, also to add `for &x in ... do` but I think that will have to wait until F# vNext since it's going to be fiddly (here `&x` is a pattern).

**Comment by cartermp** [comment]
This will also require a corresponding RFC update

**Comment by cartermp** [comment]
This is fixed and in with #5285.
